Khan/wonder-blocks
# 3/4 [feature:popover] WB-605: Implement Popover

## Summary

Renders a fully integrated Popover component.

### TODOs
This will go in the last PR (4/4)
- Add focus management
- Add keyboard support
- Add ARIA attributes

### NOTE: Stacked PRs

This is part of [WB-579: Implement Popovers](https://khanacademy.atlassian.net/browse/WB-579)

These PRs are created to cover the whole Popover implementation:

1. Implement PopoverContentCore (current PR): Renders popover content.
2. Implement PopoverContent: Add text-only, icon and illustration variations.
3. **Implement Popover: Adds Popover component to handle popover triggering and positioning. Also, integrates TooltipTail and TooltipPopper.**
4. Popover: Add Focus management and accessibility (keyboard support, ARIA attributes)

## Implementation

### Popover
[Implementation Spec](https://khanacademy.atlassian.net/wiki/spaces/FRONTEND/pages/115310606/Popover+v1+Implementation+Spec#Popover-public-)

#### Added
- `Popover`: It's used as the popover launcher.
    - Included TooltipPopper and TooltipTail from `wonder-blocks-tooltip`.
- `PopoverAnchor`: Works as trigger element and/or placement ref.
- `PopoverDialog`: Wraps the popover content and `TooltipTail`.
- `PopoverContext`: Handles `onClose` and `placement` behavior across the component.

#### Changed
- Updated `PopoverContent` and `PopoverContentCore` to make them work with the `Popover`.
- Improved documentation.
- Changed `color` prop, to use `wonder-blocks-color` palette instead of just 3 different colors.


### Tooltip
- Renamed `TooltipBubbleProps` to `PopperElementProps` to make the name more generic.
- Added `PopperElementProps` type as part of the public API.
- Added `TooltipPopper` and `TooltipTail` to the public API.
- `TooltipTail`: Added `color` prop to change the tail's bg color.

## Testplan
1. Open https://deploy-preview-440--wonder-blocks.netlify.com/#popover
2. Make sure all the examples are displaying properly.
![popover-example](https://user-images.githubusercontent.com/843075/60301953-92981180-9900-11e9-8c90-4dd28f660f8c.png)
